pkg/storage/unified/resource/kv.go

```diff
@@ -0,0 +1,213 @@
+package resource
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"iter"
+	"time"
+
+	badger "github.com/dgraph-io/badger/v4"
+)
+
+var ErrNotFound = errors.New("key not found")
+
+type SortOrder int
+
+const (
+	SortOrderUndefined SortOrder = iota
+	SortOrderAsc
+	SortOrderDesc
+)
+
+type ListOptions struct {
+	Sort     SortOrder // sort order of the results. Default is SortOrderAsc.
+	StartKey string    // lower bound of the range, included in the results
+	EndKey   string    // upper bound of the range, excluded from the results
+	Limit    int64     // maximum number of results to return. 0 means no limit.
+}
+
+// KVReader represents a key-value pair with streaming value access
+type KVReader struct {
+	Key    string
```

Comment:
We should document that this key does NOT contain section. (We could include `Section` in separate field, although that seems unnecessary)